### PR TITLE
Bugfix FXIOS-9492 Sec bugzilla 1863832 take 2

### PR DESCRIPTION
**Pull request by razvanlitianu**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9492)

## :bulb: Description

Prevent Focus from opening deeplinks from links

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: focus-ios/Blockzilla/Modules/WebView/LegacyWebViewController.swift [hunk 1]
```diff
@@ -455,6 +455,12 @@ extension LegacyWebViewController: WKNavigationDelegate {
             break
         }
 
+        // Prevent Focus from opening deeplinks from links
+        if let scheme = navigationAction.request.url?.scheme, scheme == AppInfo.appScheme {
```
Comment (mattreaganmozilla) [inline focus-ios/Blockzilla/Modules/WebView/LegacyWebViewController.swift:459]: Correct me if I'm wrong but I believe `scheme` is not required to be lower-cased, meaning that we should either do a case-insensitive string comparison here, or check against `scheme.lowercased()` etc.

```
(lldb) po url
▿ Optional<URL>
  ▿ some : hTtPs://mozilla.com
    - _url : hTtPs://mozilla.com
(lldb) po scheme
▿ Optional<String>
  - some : "hTtPs"
```

Comment (razvanlitianu) [reply focus-ios/Blockzilla/Modules/WebView/LegacyWebViewController.swift]: Updated

**Review by mattreaganmozilla — APPROVED** [review]
Would probably be good to get @nbhasin2 's eyes on this also but LGTM

**Review by nbhasin2 — APPROVED** [review]

**Comment by razvanlitianu** [comment]
@Mergifyio backport release/v131

**Pull request merged**